debsources/query.py

```diff
@@ -122,9 +122,14 @@ def location_get_path_links(endpoint, path_to):
     # this method)
     from flask import url_for
 
-    for (i, p) in enumerate(path_dict):
-        pathl.append((p, url_for(endpoint,
-                                 path_to='/'.join(path_dict[:i + 1]))))
+    if endpoint == '.versions':
+        for (i, p) in enumerate(path_dict):
+            pathl.append((p, url_for(endpoint,
+                                     packagename='/'.join(path_dict[:i + 1]))))
+    else:
+        for (i, p) in enumerate(path_dict):
+            pathl.append((p, url_for(endpoint,
+                                     path_to='/'.join(path_dict[:i + 1]))))
```

Comment:
I'll need to dive deeper in the issue, because I don't have an overview of the current state, but having 2 different ways of handling this (path_to and packagename/version) according to the endpoint will lead to more and more confusion IMO.